### PR TITLE
fix document out of sync

### PR DESCRIPTION
**Pull request by elebeaup**
Fix autocompletion when multiple content changes are received in the same event

### CODE REVIEW

File: server/src/main/java/org/rri/ideals/server/ManagedDocuments.java [hunk 1]
```diff
@@ -79,7 +78,10 @@ public void startManaging(@NotNull TextDocumentItem textDocument) {
 */
       });
 
-      docs.put(path, new VersionedTextDocumentIdentifier(uri, textDocument.getVersion()));
+      var docVersion = Optional.ofNullable(textDocument.getVersion())
```
Comment (serganch) [inline server/src/main/java/org/rri/ideals/server/ManagedDocuments.java:81]: seems like `ofNullable()` is redundant for a value of a primitive type

File: server/src/main/java/org/rri/ideals/server/ManagedDocuments.java [hunk 2]
```diff
@@ -139,7 +141,7 @@ public void updateDocument(@NotNull DidChangeTextDocumentParams params) {
       }
 
       try {
-        applyContentChangeEventChanges(doc, sortContentChangeEventChanges(contentChanges));
+        applyContentChangeEventChanges(doc, contentChanges);
```
Comment (serganch) [inline server/src/main/java/org/rri/ideals/server/ManagedDocuments.java:144]: Does this mean changes come already sorted?

Comment (elebeaup) [reply server/src/main/java/org/rri/ideals/server/ManagedDocuments.java:144]: I didn't find any information in the official documentation but it seems to be the case when I debug the application or if I look how it works on other lsp server (e.g. https://github.com/ fwcd/kotlin-language-server/blob/24c2e410e4718e79c5cbf79ec1dfe60161ff338c/server/src/main/kotlin/org/javacs/kt/SourceFiles.kt#L100)

**Review by serganch — APPROVED** [review]
LGTM, thanks!

**Pull request merged**